cray_product_catalog/catalog_delete.py

```diff
@@ -167,15 +185,43 @@ def main():
     PRODUCT_VERSION = os.environ.get("PRODUCT_VERSION").strip()  # required
     CONFIG_MAP = os.environ.get("CONFIG_MAP", "cray-product-catalog").strip()
     CONFIG_MAP_NS = os.environ.get("CONFIG_MAP_NAMESPACE", "services").strip()
+    PRODUCT_CONFIG_MAP = format_product_cm_name(CONFIG_MAP, PRODUCT)
     KEY = os.environ.get("KEY", "").strip() or None
+    MAX_RETRIES = 100
+    MAX_RETRIES_FOR_PROD_CM = 10
 
-    args = (CONFIG_MAP, CONFIG_MAP_NS, PRODUCT, PRODUCT_VERSION, KEY)
-    LOGGER.info(
-        "Removing from ConfigMap=%s in namespace=%s for %s/%s (key=%s)",
-        *args
-    )
     load_k8s()
-    modify_config_map(*args)
+
+    if KEY:
+        if KEY in CONFIG_MAP_FIELDS:
+            args = (CONFIG_MAP, CONFIG_MAP_NS, PRODUCT, PRODUCT_VERSION, MAX_RETRIES, KEY)
+        elif KEY in PRODUCT_CM_FIELDS:
+            args = (PRODUCT_CONFIG_MAP, CONFIG_MAP_NS, PRODUCT, PRODUCT_VERSION, MAX_RETRIES_FOR_PROD_CM, KEY)
+        else:
+            LOGGER.error(
+                "Invalid KEY=%s is input so exiting...",
+                KEY
+            )
+            return
+        LOGGER.info(
+            "Removing from config_map=%s in namespace=%s for %s/%s (key=%s)",
```

Comment:
If you don't change the order of the arguments in the function call,. then this LOGGER call needs to be changed, since it currently will be logging the new max_attempts argument as the key in the log message.